docs/sdks/survey-sdk/responses-polling.md

```diff
@@ -0,0 +1,95 @@
+# Responses Polling
+
+Engage Digital will poll your survey responses by performing a `GET /responses` on your bridge every 15 minutes along with a `since` parameter corresponding to the the most recent response fetched's `submitted_at` field.
+
+This parameter will be sent as milliseconds in utc time (e.g. if **base URL of the bridge** is `https://survey.bridge.com` and your most recent response was submitted on `2022-01-10T10:00:00.000Z` then Engage Digital will perform a `GET https://survey.bridge.com/responses?since=1641808800000`).
+
+## Response example
+
+Here's an example of a valid response to the responses polling:
+```json
+{
+  "id": "12345",
+  "records": [
+    {
+      "id": "12345_response_1",
+      "submitted_at": "2022-01-10T10:00:00.000Z",
+      "mapping_key": "response_mapping_key",
+      "questions": [
+        {
+          "id": "main_question",
+          "replies": [
+            {
+              "value": "5"
+            }
+          ]
+        },
+        {
+          "id": "multiple_choices_question",
+          "replies": [
+            {
+              "value": "Faster response time"
+            },
+            {
+              "value": "Better follow-up"
+            }
+        ]
+        },
+        {
+          "id": "free_text_question",
+          "replies": [
+            {
+              "value": "Everything's already perfect"
+            }
+          ]
+        },
+        ...
+      ]
+    },
+    ...
+  ]
+}
+```
+
+
+### Detailed response format
+
+| Field | Type | Mandatory | Description |
+|-|-|-|-|
+| id | String | NO | Unique identifier of the survey. |
+| records | Array | **YES** | List of responses, see [Response object description](#response-object-description). |
+
+
+### Response object description
+
+| Field | Type | Mandatory | Description |
+|-|-|-|-|
+| id | String | **YES** | Unique identifier of the response. |
+| submitted_at | Time | NO | Time the response was submitted at. |
```

Comment:
We're saying we'll use this field to add a `since` param, but this field isn't mandatory? What happens if the bridge never sends us this field, we'll always poll every responses?

Reply:
the field isn't mandatory in ED's code but we might wanna make it mandatory as without it stats will be all messed up

I guess we never made it mandatory as Alchemer always sends it and we trusted them but now that we're allowing pretty much everyone to send responses we may want to make it mandatory or at least ignore the response if there's no `submitted_at`, WDYT?

Reply:
What you're suggesting is the same thing actually, because if it's mandatory and not present, we ignore the response. We won't raise.

So yes, let's do it please. (mandatory in doc + ignore if blank in code)

Reply:
my "proposal" wasn't really clear but there's a slight difference:

1. when I was talking about making it mandatory I meant adding a presence validation in the model -> will apply to all `Response` no matter the supplier
2. only ignoring response without `submitted_at` in survey sdk that would only handle this case for survey sdk but if we ever add another supplier we'll have to either trust them or ignore the response if there's no `submitted_at`

my vote goes to #1 as a response without a `submitted_at` doesn't really make sense to me no matter the supplier

Reply:
Yes ok, just need to ensure that current data won't break, and also response fetching don't break, as usual. This can be done in another ticket though, and just need to change to Mandatory in this doc

Reply:
done in d7bc181 and I created RD-21074 to handle the code change